### ISSUE TITLE
Add Title attribute to Button block

### ISSUE THREAD

**Issue by jamieschmid** [WordPress/gutenberg#69032]
## What problem does this address?
Addresses accessibility issues for screen readers and crawlers when using anchor tags without descriptive link copy. Allows users to use copy like "Click here" while still remaining accessible and descriptive for screen readers and crawlers.

## What is your proposed solution?
Add a field for Link Title in the Advanced tab. It already has name (HTML anchor) and Rel. It is only missing Title.

**Comment by nirav7707** [comment]
Hello @jamieschmid Thanks! Could you please share an example where the copy of the link is not descriptive?

**Comment by t-hamano** [comment]
> Allows users to use copy like "Click here" while still remaining accessible and descriptive for screen readers and crawlers.

While I understand that in practice there are cases like this, the Accessibility Handbook states the following:

https://make.wordpress.org/accessibility/handbook/best-practices/quick-start-guide/#links

> - Use meaningful link text to describe the link: Never just “click here” or “more info”. The text of the link should indicate what clicking the link does. (Screen reader users often use a list of links out of context.)
> - Avoid the `title` attribute in anchor elements (`<a>`).

Therefore, I am concerned about whether this rule applies to the block editor.

cc @afercia @alexstine @joedolson

**Comment by afercia** [comment]
A link text like 'Click here' or similar isn't meaningful enough for accessibility, especially when read out of context. It isn't great for SEO as well. The recommendation is to always use meaningful link text. Worth also reminding the title attribute is only accessible to some users. As such, though it is used to compute the accessible description, it isn't recommended. WordPress has been progressively removing the usage of the title attribute for good reasons, see https://core.trac.wordpress.org/ticket/24766

**Comment by joedolson** [comment]
I see little reason to support a mechanism to add title attributes in any context, given that they have minimal benefit and some negative impacts. If somebody wants improved accessibility, they can and should accomplish that by re-writing their link text.

**Comment by t-hamano** [comment]
Thank you everyone for your feedback. Let's close this issue as "Not planned".

@jamieschmid, if you have any comments, please feel free to comment.
